crates/oxc_semantic/src/node.rs

```diff
@@ -118,6 +118,10 @@ impl<'a> AstNodes<'a> {
         &mut self.nodes[ast_node_id]
     }
 
+    /// Walk up the AST, iterating over each parent node.
+    ///
+    /// The first node produced by this iterator is the first parent of the node
+    /// pointed to by `node_id`. The last node will usually be a `Program`.
```

Comment:
usually? when would it not be?

Reply:
I'm not sure. I don't want to say for certain it won't be, because I don't know.